spec/matchers/be_called_with_arguments_spec.rb

```diff
@@ -26,6 +27,37 @@
       it 'correctly matches when wildcard is used for all arguments' do
         expect(@command).to be_called_with_arguments(anything, anything)
       end
+
+      it 'displays the diff between what was called and what was expected' do
+        begin
+          expect(@command).to be_called_with_arguments('not_first_argument', 'second_argument')
+        rescue RSpec::Expectations::ExpectationNotMetError => rex
+          error_string = <<-multiline_string
+expected stubbed_command to be called with arguments ["not_first_argument", "second_argument"]
+Diff:
+@@ -1,2 +1,3 @@
+-["not_first_argument", "second_argument"]
++["first_argument", "second_argument"]
++["third_argument", "fourth_argument"]
+multiline_string
+          expect(rex.message.uncolorize).to eql error_string
+        end
+      end
+
+      it 'displays the diff between what was called and what was expected (negative case)' do
+        begin
+          expect(@command).to_not be_called_with_arguments('first_argument', 'second_argument')
+        rescue RSpec::Expectations::ExpectationNotMetError => rex
+          expected_error_string = <<-multiline_string
+expected stubbed_command not to be called with arguments ["first_argument", "second_argument"]
+Diff:
+@@ -1,2 +1,3 @@
+ ["first_argument", "second_argument"]
++["third_argument", "fourth_argument"]
```

Comment:
Hmmmm this is kinda weird, I don't even know what I would expect it to show in the negative case. Why does it show this? `["third_argument", "fourth_argument"]` isn't even part of the test here (we're just testing that `["first_argument", "second_argument"]` get called. I'm not saying this isn't the best thing to output to the user, I just want you to convince me that it is :)